src/components/views/elements/AppTile.js

```diff
@@ -51,42 +52,60 @@ export default React.createClass({
         creatorUserId: React.PropTypes.string,
     },
 
-    getDefaultProps: function() {
+    getDefaultProps() {
         return {
             url: "",
         };
     },
 
-    getInitialState: function() {
-        const widgetPermissionId = [this.props.room.roomId, encodeURIComponent(this.props.url)].join('_');
+    /**
+     * Set initial component state when the App wUrl (widget URL) is being updated
+     * @param  {Object} props The component props *must* be passed (rather than using this.props) so that it can be called with future props, e.g. from componentWillReceiveProps.
+     * @return {Object} Updated component state to be set with setState
+     */
+    _getNewUrlState(props) {
+        const widgetPermissionId = [props.room.roomId, encodeURIComponent(props.url)].join('_');
         const hasPermissionToLoad = localStorage.getItem(widgetPermissionId);
         return {
-            loading: false,
-            widgetUrl: this.props.url,
+            initialising: true,   // True while we are mangling the widget URL
+            loading: true,        // True while the iframe content is loading
+            widgetUrl: props.url,
             widgetPermissionId: widgetPermissionId,
             // Assume that widget has permission to load if we are the user who added it to the room, or if explicitly granted by the user
-            hasPermissionToLoad: hasPermissionToLoad === 'true' || this.props.userId === this.props.creatorUserId,
+            hasPermissionToLoad: hasPermissionToLoad === 'true' || props.userId === props.creatorUserId,
```

Comment:
I think it'd be clearer to do the comparison to the string at the point you take this out of localStorage.